### ISSUE TITLE
NPE on updateSatelliteStatus

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#13955]
### Describe your problem!

User reports constant crashing with current beta 

### How to reproduce?

Unclear

### Actual result after these steps?

Logifle shows many time this:

```
02-01 09:22:08.390 19128 19128 E AndroidRuntime: FATAL EXCEPTION: main
02-01 09:22:08.390 19128 19128 E AndroidRuntime: Process: cgeo.geocaching, PID: 19128
02-01 09:22:08.390 19128 19128 E AndroidRuntime: io.reactivex.rxjava3.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NullPointerException: Attempt to invoke virtual method 'void cgeo.geocaching.ui.LocationStatusView.updateSatelliteStatus(cgeo.geocaching.sensors.GnssStatusProvider$Status)' on a null object reference
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:718)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:715)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.observers.LambdaObserver.onNext(LambdaObserver.java:67)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:202)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:256)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:123)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8167)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void cgeo.geocaching.ui.LocationStatusView.updateSatelliteStatus(cgeo.geocaching.sensors.GnssStatusProvider$Status)' on a null object reference
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at cgeo.geocaching.CompassActivity$2.accept(CompassActivity.java:304)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at cgeo.geocaching.CompassActivity$2.accept(CompassActivity.java:301)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	at io.reactivex.rxjava3.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
02-01 09:22:08.390 19128 19128 E AndroidRuntime: 	... 10 more
```

### Reproducible

Unclear

### c:geo Version

2023.01.30-RC

### System information

```text
c:geo version: 2023.01.30-RC

Device:
-------
- Device type: SM-G960F (starltexx, samsung)
- Available processors: 8
- Android version: 10
- Android build: QP1A.190711.020.G960FXXUHFVG4
- Screen resolution: 1080x2076px (360x668dp)
- Pixel density: 3.0
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 23.03.13 (100408-503260631)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:187
- Set language: en_NZ (system default)
- System date format: d/MM/yy
- Time zone: GMT+13:00
- Debug mode active: no
- Last backup: never
- Routing mode: Off
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: NewZealand (OAM) (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:cgeo/maps/NewZealand (OAM).map
  - Atts: Maps: (c)Christian Kernbeis & Michael Bechtold , Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2022
  - Theme: Elevate.zip:Elements.xml

Filters:
-------
- Hide waypoints: original parking visited
- LIVE: Status: 3 items ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status:owned_no:found_no:exclude_disabled))
- OFFLINE: None ([:inconclusive=false:advanced=true])

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
   Geocaching.com Adventure Lab
- Geocaching.com date format: dd/MM/yyyy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2023-02-03 14-52)
- Last successful login on geocaching.com: 2023-02-04 11-26
- Routing: external / BRouter installed: false
- Installed c:geo plugins:  none

Permissions
-------
-android.permission.ACCESS_COARSE_LOCATION:: granted
-android.permission.ACCESS_FINE_LOCATION:: granted
-android.permission.READ_CONTACTS:: DENIED
-android.permission.READ_EXTERNAL_STORAGE:: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (12.2 GB free) v3 internal isDir(10 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (12.2 GB free) v3 internal isDir(10 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (12.2 GB free) v3 external non-removable isDir(846 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (12.2 GB free) v3 internal isDir(0 entries)
- Public Folders: #12
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=49, dirs:>=9, totalFileSize:>=1.8 GB, free space: 12.2 GB, files on device: 1740800)
  - OFFLINE_MAPS: /cgeo/maps (User-Defined)[/cgeo/maps[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps/document/primary%3Acgeo%2Fmaps, Av:true, files:12, dirs:2, totalFileSize:1.8 GB, free space: 12.2 GB, files on device: 1740800)
  - OFFLINE_MAP_THEMES: /cgeo/maps/_themes (Default)[/cgeo/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps::/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps/document/primary%3Acgeo%2Fmaps%2F_themes, Av:true, files:4, dirs:0, totalFileSize:504.6 KB, free space: 12.2 GB, files on device: 1740800)
  - OFFLINE_MAP_SHADING: /cgeo/maps/_hgt (Default)[/cgeo/maps/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps::/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps/document/primary%3Acgeo%2Fmaps%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 12.2 GB, files on device: 1740800)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:30, dirs:0, totalFileSize:988.9 KB, free space: 12.2 GB, files on device: 1740800)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:5, dirs:0, totalFileSize:2.5 MB, free space: 12.2 GB, files on device: 1740800)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=9, dirs:>=6, totalFileSize:>=134.5 MB, free space: 12.2 GB, files on device: 1740800)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:1, dirs:0, totalFileSize:0 B, free space: 12.2 GB, files on device: 1740800)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:1, dirs:0, totalFileSize:0 B, free space: 12.2 GB, files on device: 1740800)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81.2 KB, free space: 12.2 GB, files on device: 1740800)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 12.2 GB, files on device: 1740800)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 12.2 GB, files on device: -1)
- Map render theme path: Elevate.zip:Elements.xml
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #2
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (22 Mar 2021, 10:54 am):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fmaps (22 Mar 2021, 10:54 am):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:34.3 MB) on system internal storage

--- End of system information ---
```

### Additional Information

Support ticker 281621

**Comment by eddiemuc** [comment]
Root cause for this is same as #13921, the problem will be fixed with PR #13948

As intermediate measurement: compass activity should work without crash as long as only portrait mode is used (do not use landscape)

**Comment by eddiemuc** [comment]
I believe this is fixed
